mongodb/configure.sls

```diff
@@ -0,0 +1,116 @@
+{% from "mongodb/map.jinja" import mongodb with context %}
+
+include:
+  - .service
+
+copy_mongodb_key_file:
+  file.managed:
+    - name: {{ mongodb.cluster_key_file }}
+    - contents: "{{ salt.pillar.get('mongodb:cluster_key') }}"
+    - owner: mongodb
+    - group: mongodb
+    - mode: 0600
+
+place_mongodb_config_file:
+  file.managed:
+    - name: /etc/mongodb.conf
+    - template: jinja
+    - source: salt://mongodb/templates/mongodb.conf.j2
+    - require:
+      - file: copy_mongodb_key_file
+    - require_in:
+      - service: mongodb_service_running
+    - watch_in:
+      - service: mongodb_service_running
+
+wait_for_mongo:
+  cmd.run:
+    - name: |
+        until [ `netstat -tunlp | grep {{ mongodb.port }} | wc -l` -eq 1 ]
```

Comment:
We might want to use -ge instead of -eq in case it is listening on both IPv4 and IPv6 interfaces.